Makefile

```diff
@@ -0,0 +1,26 @@
+all: build
+.PHONY: all
+
+# Include the library makefile
+include $(addprefix ./vendor/github.com/openshift/library-go/alpha-build-machinery/make/, \
+	golang.mk \
+	targets/openshift/deps.mk \
+	targets/openshift/images.mk \
+)
+
+# This will call a macro called "build-image" which will generate image specific targets based on the parameters:
+# $0 - macro name
+# $1 - target suffix
+# $2 - Dockerfile path
+# $3 - context directory for image build
+# It will generate target "image-$(1)" for builing the image an binding it as a prerequisite to target "images".
+$(call build-image,origin-node,./images/sdn/Dockerfile,.)
+$(call build-image,origin-network-controller,./images/network-controller/Dockerfile,.)
```

Comment:
Neither of these image names is really right; the sdn node image is currently called "origin-node", but that's a historical accident. (The SDN binary is the only thing left in the old `openshift start node` image.) It should be renamed to "origin-sdn" or "origin-openshift-sdn" while we're moving it.

For the controller side, "origin-network-controller" makes it sound generically network-related, but it's not; it's very specifically the controller for just the openshift-sdn network plugin. (It doesn't get run if you have any other network plugin configured.) So it should be "origin-sdn-controller" or "origin-openshift-sdn-controller".

Reply:
(comments about "network controller" vs "sdn controller" apply to the labels inside the image too, though those are easier to fix later without needing to coordinate multiple repos).

Also, you could just merge them into a single image if that made sense...

Reply:
-node is legacy, we aren't planning to change that as part of this push.


I suspect as a post-step, you'll want a single unified sdn image, but I think that can happen just as easily afterwards.  This is about seeing if we can get fast enough responses on images to make your CI a little more effective.

Reply:
I agree with the rename of the node image. But let's leave that for a follow-up. We cannot safe the world in one PR. Better step by step.

@stlaz see network-controller name.

@danwinship is that a definitive answer for the network-controller image name?

Reply:
we need to change it in CI as that's where  the definitive name for CI resides. "origin-sdn-controller" or "origin-openshift-sdn-controller"?

Reply:
from Slack discussion with @danwinship: sdn-controller (with automatic ose/origin prefix)